src/H5VLcallback.c

```diff
@@ -3551,27 +3535,21 @@ H5VL__file_open_find_connector_cb(H5PL_type_t plugin_type, const void *plugin_in
 
     assert(udata);
     assert(udata->filename);
-    assert(udata->connector_prop);
     assert(cls);
     assert(plugin_type == H5PL_TYPE_VOL);
 
-    /* Silence compiler */
-    (void)plugin_type;
-
-    udata->cls = cls;
```

Comment:
Simplified this routine a bit, and replaced the H5E_clear_stack call with a pause/resume on the error stack.